config/size-report.js

```diff
@@ -0,0 +1,217 @@
+const path = require('path');
+const fs = require('fs');
+const util = require('util');
+
+const gzipSize = require('gzip-size');
+const fetch = require('node-fetch');
+const prettyBytes = require('pretty-bytes');
+const escapeRE = require('escape-string-regexp');
+const chalk = new require('chalk').constructor({ level: 4 });
+
+const readdir = util.promisify(fs.readdir);
+const stat = util.promisify(fs.stat);
+
+function getTravis(path) {
+  return fetch('https://api.travis-ci.org' + path, {
+    headers: { 'Travis-API-Version': '3' },
+  });
+}
+
+function getTravisJson(path) {
+  return getTravis(path).then(r => r.json());
+}
+
+function getTravisText(path) {
+  return getTravis(path).then(r => r.text());
+}
+
+/**
+ * Recursively-read a directory and turn it into an array of { name, size, gzipSize }
+ */
+async function dirToInfoArray(startPath, {
```

Comment:
Maybe rely on an existing impl here? Like [`readdirp`](https://www.npmjs.com/package/readdirp)?

Reply:
Yeah I'll give it a shot.